ruma-appservice-api/src/lib.rs

```diff
@@ -4,6 +4,69 @@
 
 #![warn(missing_debug_implementations, missing_docs)]
 
+use serde::{Deserialize, Serialize};
+
 pub mod event;
 pub mod query;
 pub mod thirdparty;
+
+/// A namespace defined by an application service.
```

Comment:
Would be good to have a link to https://matrix.org/docs/spec/application_service/r0.1.2#registration in the extended docs (i.e. not part of the tagline):

```suggestion
/// A namespace defined by an application service.
///
/// Used for [appservice registration](https://matrix.org/docs/spec/application_service/r0.1.2#registration).
```

Same for the other types added by this PR.